kpops/components/base_components/models/from_section.py

```diff
@@ -19,24 +20,39 @@ class Config(BaseConfig):
         use_enum_values = True
 
     @root_validator
-    def extra_topic_role(cls, values):
+    def extra_topic_role(cls, values: dict) -> dict:
         is_extra_topic = values["type"] in (
             InputTopicTypes.EXTRA,
             InputTopicTypes.EXTRA_PATTERN,
         )
         if is_extra_topic and not values.get("role"):
             raise ValueError(
-                "If you define an extra input topic or extra input pattern, you have to define a role."
+                "If you define an extra input component, extra input topic, or extra input pattern, you have to define a role."
             )
         if not is_extra_topic and values.get("role"):
             raise ValueError(
-                "If you do not define an input topic or input pattern, the role is unnecessary."
+                "If you do not define an input component, input topic, or input pattern, the role is unnecessary."
             )
         return values
 
 
+class FromComponent(FromTopic):
+    pass
+
+
+TopicName = NewType("TopicName", str)
+ComponentName = NewType("ComponentName", str)
+
+
 class FromSection(BaseModel):
-    topics: dict[str, FromTopic]
+    topics: dict[TopicName, FromTopic] = Field(
+        default={},
+        description="Topics to read from.",
+    )
+    components: dict[ComponentName, FromComponent] = Field(
```

Comment:
Why aren't you using FromTopic here?

Reply:
you're right, I wasn't sure if I was gonna add anything else to it initially